Support for GEPRCF745BTHD (#27491)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/GEPRCF745BTHD/README.md

# GEPRC TAKER F745 BT Flight Controller

The TAKER F745 BT is a flight controller produced by [GEPRC](https://geprc.com/).

## Features

 - STM32F745 microcontroller
 - MPU6000+ICM42688 dual IMU
 - BMP280 barometer
 - microSD based 512MB flash logging
 - AT7456E OSD
 - 7 UARTs
 - 8 PWM outputs

## Pinout

![TAKER F745 BT Board](TAKER_F745_BT_Board_Top.jpg "GEPRCF745BTHD")
![TAKER F745 BT Board](TAKER_F745_BT_Board_Bottom.jpg "GEPRCF745BTHD")

## UART Mapping

The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
receive pin for UARTn. The Tn pin is the transmit pin for UARTn.

 - SERIAL0 -> USB

>  - SERIAL1 -> UART1 (DJI-VTX, DMA-enabled)

Reviewer: ```suggestion
 - SERIAL1 -> UART1 (DisplayPort, DMA-enabled)
```

Reviewer 2: Fixed

 - SERIAL2 -> UART2 (RCIN, DMA-enabled) 
 - SERIAL3 -> UART3 (connected to internal BT module, not currently usable by ArduPilot)
 - SERIAL4 -> UART4 (GPS)

>  - SERIAL6 -> UART6 (leisure)

Reviewer: ```suggestion
 - SERIAL6 -> UART6 (USER)
```

Reviewer 2: Fixed

>  - SERIAL7 -> UART7 (leisure)

Reviewer: ```suggestion
 - SERIAL7 -> UART7 (USER)
```

Reviewer 2: Fixed

 - SERIAL8 -> UART8 (ESC Telemetry)

## RC Input

> RC input is configured on the R2 (UART2_RX) pin. It supports all RC protocols.

Reviewer: ```suggestion
RC input is configured by default via the USAR2 RX input. It supports all unidirectional RC protocols except PPM. FPort and full duplex protocols. like CRSF/ELRS, will need to use TX2 also.

Note:
If the receiver is FPort or a full duplex protocol, then the receiver must be tied to the USART2 TX pin and [SERIAL2_OPTIONS](https://ardupilot.org/copter/docs/parameters.html#serial1-options) = 7 (invert TX/RX, half duplex), and [RSSI_TYPE](https://ardupilot.org/copter/docs/parameters.html#rssi-type) =3.
```

Reviewer 2: Fixed

 
## FrSky Telemetry
 
FrSky Telemetry is supported using the Tx pin of any UART including SERIAL2/UART2. You need to set the following parameters to enable support for FrSky S.PORT (example shows SERIAL3).
 
  - SERIAL3_PROTOCOL 10
  - SERIAL3_OPTIONS 7
  
## OSD Support

> The TAKER F745 BT supports OSD using OSD_TYPE 1 (MAX7456 driver).

Reviewer: ```suggestion
The TAKER F745 BT supports analog OSD using its internal OSD chip and simultaneously HD goggle DisplayPort OSDs via the HD VTX connector.
```

Reviewer 2: Fixed

## VTX Support

> The SH1.0-6P connector supports a standard DJI HD VTX connection. Pin 1 of the connector is 12v so be careful not to connect

Reviewer: ```suggestion
The SH1.0-6P connector supports a standard DJI HD VTX connection. Pin 1 of the connector is 12v (or VBAT by solder pad selection)so be careful not to connect
```

Reviewer: I dont know what the Chinese says for the VTX pin...please translate...also what is the difference between the VSW and this pin? do you have a power switch onboard? its not included in the hwdef...

## PWM Output

The TAKER F745 BT supports up to 9 PWM outputs. The pads for motor output

> M1 to M4 on the motor connector, plus M9 for LED strip or another

Reviewer: ```suggestion
M1 to M4 are on the esc connector, M5- M8 are solder pads, plus M9  is defaulted for serial LED strip or can be used as another
```

Reviewer 2: Fixed

PWM output.

The PWM is in 4 groups:

 - PWM 1-4 in group1
 - PWM 5-6 in group2
 - PWM 7-8 in group3
 - PWM 9 in group4

Channels within the same group need to use the same output rate. If
any channel in a group uses DShot then all channels in the group need

> to use DShot. Channels 1-4 support bi-directional DShot.

Reviewer: ```suggestion
to use DShot. Channels 1-8 support bi-directional DShot.
```

Reviewer 2: Not true - only 1-4

Reviewer: my mistake....cross eyes

## Battery Monitoring

The board has a internal voltage sensor and connections on the ESC connector for an external current sensor input.
The voltage sensor can handle up to 6S.
LiPo batteries.

> The correct battery setting parameters are:

Reviewer: ```suggestion
The default battery setting parameters are:
```

Reviewer 2: Fixed

 - BATT_MONITOR 4
 - BATT_VOLT_PIN 13
 - BATT_VOLT_SCALE 11.0
 - BATT_CURR_PIN 12

>  - BATT_CURR_SCALE 58.8

Reviewer: ```suggestion
 - BATT_CURR_SCALE 28.5
```
this is the hwdef value

Reviewer 2: Fixed
